diff --git a/arch/arm/boot/dts/sun4i-a10-cubieboard.dts b/arch/arm/boot/dts/sun4i-a10-cubieboard.dts
index 6ca02e8..045adac 100644
--- a/arch/arm/boot/dts/sun4i-a10-cubieboard.dts
+++ b/arch/arm/boot/dts/sun4i-a10-cubieboard.dts
@@ -233,7 +233,17 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&spi0_pi_pins>,
 		    <&spi0_cs0_pi_pin>;
+	id = <0>;
 	status = "okay";
+
+	spidev0_0: spidev0@0 {
+		compatible = "userspi";
+		reg = <0>;
+		spi-max-frequency = <8000000>;
+		/* spi-cpha; sets CPHA=1, default is CPHA=0 */
+		/* spi-cpol; sets CPOL=1, default is CPOL=0 */
+		/* spi-cs-high; default is spi cs low */
+	};
 };
 
 &uart0 {
